src/encoded/static/components/visualizations.js

```diff
@@ -793,205 +796,175 @@ export class ChartTable extends React.Component {
-        } else {
-            modifiedSelectedStates = [...this.state.selectedStates, clickID];
-        }
-        this.setState(prevState => ({
-            selectedStates: modifiedSelectedStates,
-            filteredData: prevState.data.filter((d) => {
-                const searchTerm = String(sanitizedString(this.state.unsanitizedSearchTerm));
-                if (searchTerm === '') {
-                    return (modifiedSelectedStates.includes(sanitizedString(lookupChromatinNames(d.value))));
-                }
-                if (d.biosample_ontology.term_name) {
-                    return (sanitizedString(lookupChromatinNames(d.value)).includes(searchTerm) || sanitizedString(d.biosample_ontology.term_name).includes(searchTerm) || sanitizedString(d.biosample_ontology.organ_slims.join(', ')).includes(searchTerm)) && modifiedSelectedStates.includes(sanitizedString(lookupChromatinNames(d.value)));
-                }
-                return (sanitizedString(lookupChromatinNames(d.value)).includes(searchTerm)) && modifiedSelectedStates.includes(sanitizedString(lookupChromatinNames(d.value)));
-            }),
-        }));
-    }
-
-    handleSearch(event) {
-        // Unsanitized search term entered by user for display
-        this.setState({ unsanitizedSearchTerm: event.target.value });
-        // Search term entered by the user
-        const filterVal = String(sanitizedString(event.target.value));
-        let filteredData = [];
-        if (filterVal === '') {
-            filteredData = this.state.data;
-        } else {
-            filteredData = this.state.data.filter((d) => {
-                if (d.biosample_ontology.term_name) {
-                    return sanitizedString(lookupChromatinNames(d.value)).includes(filterVal) || sanitizedString(d.biosample_ontology.term_name).includes(filterVal) ||
-                    sanitizedString(d.biosample_ontology.organ_slims.join(', ')).includes(filterVal);
-                }
-                return sanitizedString(lookupChromatinNames(d.value)).includes(filterVal);
+    componentDidUpdate(prevProps) {
+        if (!(_.isEqual(prevProps.sortedKeys, this.props.sortedKeys)) && this.props.sortedKeys) {
+            this.setState({
+                chartKeys: this.props.sortedKeys,
             });
         }
-        const fakeFacets = Object.keys(this.state.chartData)
-            .reduce((obj, key) => {
-                obj[key] = 0;
-                return obj;
-            }, {});
-        const newSelectedStates = [];
-        filteredData.forEach((d) => {
-            fakeFacets[lookupChromatinNames(d.value)] += 1;
-            const chromatinValue = sanitizedString(lookupChromatinNames(d.value));
-            if (!newSelectedStates.includes(chromatinValue)) {
-                newSelectedStates.push(chromatinValue);
+        if (prevProps.chartWidth !== this.props.chartWidth) {
+            if (this.props.chartWidth <= this.state.leftMargin) {
```

Comment:
Looks like `leftMargin` and  possibly `chartKeys` don’t need to be in state, but calculated in `render()`. AFAICT `chartMax` and `chartData` don’t change values unless props changes, so they don’t need to be in state either.

Reply:
I actually think that "leftMargin" and "chartKeys" do need to be in state. I think otherwise a re-render isn't triggered when the props are updated - we really need "componentDidUpdate" to get called to update state.

Reply:
Sometimes I think that most of Regulome could stand to be re-written, maybe I'll get to it piece by piece as we redo the pages.